src/typ.rs

```diff
@@ -293,6 +293,7 @@ pub struct TypeConstructor {
     pub module: Vec<String>,
     pub parameters: Vec<Arc<Type>>,
     pub typ: Arc<Type>,
+    pub inline_to: Option<String>,
```

Comment:
Rather than adding this `inline_to` field let's reuse the existing constructor functionality. We will need to have a new constructor variant type that will generate the right Erlang, but otherwise it can be the same as normal record constructors.